src/on_demand/aca_on_demand_engine.cpp

```diff
@@ -310,8 +409,26 @@ void ACA_On_Demand_Engine::parse_packet(uint32_t in_port, void *packet)
   }
 
   if (_protocol != Protocol::Protocol_INT_MAX_SENTINEL_DO_NOT_USE_) {
-    on_demand_reply = unknown_recv(vlan_id, ip_src, ip_dest, port_src, port_dest, _protocol);
-    on_demand(on_demand_reply, in_port, packet, SIZE_ETHERNET + vlan_len + packet_size, _protocol);
+    packet_size = SIZE_ETHERNET + vlan_len + packet_size;
+    uuid_t uuid;
+    uuid_generate_time(uuid);
+    char uuid_str[37];
+    uuid_unparse_lower(uuid, uuid_str);
+    data_for_on_demand_call *data = new data_for_on_demand_call;
+    // auto size_of_packet_void_pointer = sizeof(packet);
+    void *packet_copy = malloc(packet_size);
+    memcpy(packet_copy, packet, packet_size);
+    data->in_port = in_port;
+    data->packet = packet_copy;
+    data->packet_size = packet_size;
+    data->protocol = _protocol;
+    request_uuid_on_demand_data_map[uuid_str] = data;
```

Comment:
@zzxgzgz So you put every unknown packet into a map including it's packet_payload. I am afraid of you will run out of memory if the unknown request never got processed by NCM and there are many unknown packets coming in.
We need to figure out how to handle this exception. For example, set up a threshold for each unknown packet. If threshold time-up, just drop packet and remove it from the map.

Reply:
I think we can clear the map periodically, like every one minute.

A few scenarios to think about:
1. Packet comes in, only a few entries in the map -> no problem.
2. Packet comes in, a lot of entries in the map -> means that a lot of entries haven't been taken care of, probably NCM is down, then we clear the map if the time exceeds the threshold. 
3. Packet doesn't come in, we don't touch the map.

Reply:
Added another thread to clean the `request_uuid_on_demand_data_map ` periodically.